compiler/src/compiler/string_to_rcst.rs

```diff
@@ -640,28 +640,31 @@ mod parse {
     }
 
     #[instrument(level = "trace")]
-    fn text_interpolation(input: &str, curly_brace_count: usize) -> Option<(&str, Rcst)> {
+    fn text_interpolation(
+        input: &str,
+        indentation: usize,
+        curly_brace_count: usize,
+    ) -> Option<(&str, Rcst)> {
         let (mut input, mut opening_curly_braces) =
             parse_multiple(input, opening_curly_brace, Some((curly_brace_count, true)))?;
 
-        if let Some((new_input, whitespace)) = single_line_whitespace(input) {
-            input = new_input;
-            let last = opening_curly_braces.pop().unwrap();
-            opening_curly_braces.push(last.wrap_in_whitespace(vec![whitespace]));
-        }
+        let (new_input, whitespace) = whitespaces_and_newlines(input, indentation + 1, false);
+        input = new_input;
```

Comment:
You can change this to `let (input, …) = …` to hide the previous `input` variable and avoid the additional assignment